wagtailio/core/blocks.py

```diff
@@ -179,3 +183,356 @@ class HomePageBlock(blocks.StreamBlock):
+        template = "patterns/components/streamfield/cards_list_block.html"
+        label = "Cards"
+
+
+class LogoCardsBlock(blocks.StructBlock):
+    cards = blocks.ListBlock(LogoCardBlock())
+
+
+    class Meta:
+        template = "patterns/components/streamfield/logo_cards_list_block.html"
+        label = "Logo cards"
+
+
+class ComparisonTableBlock(blocks.StructBlock):
+    comparison_table = TypedTableBlock([
+        ('rich_text', blocks.RichTextBlock(features=["bold", "italic", "link"])),
+        ('image', ImageChooserBlock()),
+    ])
+
+    def get_context(self, value, parent_context=None):
+        ctx = super().get_context(value, parent_context)
+
+        rows = []
+        columns = next(ctx["value"]["table"].rows, [])
+        num_rows = 0
+
+        for column_data in columns:
+            # column_data.value contains the cells of a single column
+            block_name = column_data.block.name
+            curr_row = 0
+            for row_data in column_data.value:
+                if num_rows <= curr_row:
+                    rows.append([])
+                    num_rows += 1
+                rows[curr_row].append({"block_name": block_name, "value": row_data})
+                curr_row += 1
+
+        ctx["value"]["rows"] = rows
+        return ctx
+
+    class Meta:
+        icon = "table"
+        template = "patterns/components/streamfield/comparison_table_block.html"
+
+
+class HeadlineBlock(blocks.StructBlock):
+    heading = blocks.CharBlock(max_length=255)
+    sub_heading = blocks.TextBlock(required=False)
+    intro = blocks.TextBlock(required=False)
+    cta = CTABlock(required=False)
+    icon = blocks.ChoiceBlock(required=False, choices=SVGIcon.choices)
+    dark_background = blocks.BooleanBlock(required=False, default=False)
+
+    class Meta:
+        icon = "title"
+        label = "Headline"
+        template = "patterns/components/streamfield/headline_block.html"
+
+
+class HighlightBlock(blocks.StructBlock):
+    heading = blocks.CharBlock(max_length=100)
+    description = blocks.TextBlock(required=False)
+    image = ImageChooserBlock()
+    image_on_right = blocks.BooleanBlock(required=False, default=False)
+    meta_text = blocks.CharBlock(required=False, max_length=50)
+    meta_icon = blocks.ChoiceBlock(required=False, choices=SVGIcon.choices)
+    cta = CTABlock(required=False)
+
+    class Meta:
+        icon = "placeholder"
+        label = "Highlight"
+        template = "patterns/components/streamfield/highlight_block.html"
+
+
+class IconBulletBlock(blocks.StructBlock):
+    icon = blocks.ChoiceBlock(choices=SVGIcon.choices)
+    heading = blocks.CharBlock(max_length=255)
+    description = blocks.RichTextBlock(
+        required=False,
+        features=["bold", "italic", "link"],
+    )
+    cta = CTABlock(required=False)
+
+    class Meta:
+        icon = "tick-inverse"
+        label = "Icon bullet"
+        template = "patterns/components/streamfield/icon_bullet_block.html"
```

Comment:
Should we be putting the streamfields in their own directory? 
```
patterns/components/streamfield/icon_bullet_block/icon_bullet_block.html
```
Each html file will have a corresponding yaml file so it would make the readability of the streamfields a bit easier.

Reply:
I noticed in #186 that the `cta` template was placed directly in the **streamfields** directory, so I followed the same pattern 🙂

Reply:
I think that may have been because `cta` was the first one to be added which should have been in its own directory. The streamfields that have been added since then have their own directory - https://github.com/wagtail/wagtail.org/tree/next/wagtailio/project_styleguide/templates/patterns/components/streamfields
I can create an MR to move `cta` into it's own dir.